doc/source/reference/arrays.classes.rst

```diff
@@ -215,6 +215,13 @@ Matrix objects
 .. index::
    single: matrix
 
+.. note::
+   It is strongly advised *not* to use the matrix subclass.  As described
+   below, it makes writing functions that deal consistently with matrices
+   and regular arrays very difficult. Currently, the main reason still to use
```

Comment:
This sentence is incomplete.

Perhaps "Currently, they are mainly used for interacting with `scipy.sparse`."